src/kOS/Utilities/VesselUtils.cs

```diff
@@ -318,7 +389,312 @@ public static object GetSolarPanelStatus(Vessel vessel)
 
         public static void SolarPanelCtrl(Vessel vessel, bool state)
         {
-            vessel.rootPart.SendEvent(state ? "Extend" : "Retract");
+            foreach (var p in vessel.parts)
+            {
+                foreach (var c in p.FindModulesImplementing<ModuleDeployableSolarPanel>())
+                {
+                    if (state) { c.Extend(); }else { c.Retract(); }
+                }
+            }
+        }
+
+        public static object GetRadiatorStatus(Vessel vessel)
+        {
+            var atLeastOneRadiator = false; // No radiators at all? Always return false
+
+            foreach (var p in vessel.parts)
+            {
+                foreach (var c in p.FindModulesImplementing<ModuleActiveRadiator>())
+                {
+                    atLeastOneRadiator = true;
+
+                    if (! c.IsCooling)
```

Comment:
My primary concern (and it applies to chutes too I think) is in the case of `toggle radiators.` where it will first check radiator status and then set it to the opposite value.  If you have a deployable radiator inside of a fairing or cargo bay, toggle will never be able to de-activate the other radiators because it will always read the current value as `false`.  I'll take a closer look at the interaction implied by the fixed radiators having an active state as well.